src/gt4sd/properties/core.py

```diff
@@ -34,6 +34,21 @@ class PropertyPredictorParameters(BaseModel):
     pass
 
 
+# Base class for property predictors that use S3 artifacts
+class S3Parameters(PropertyPredictorParameters):
+
+    algorithm_type: str = "prediction"
+
+    domain: str = Field(
```

Comment:
consider using an enum to control better the inputs.
```
from enum import Enum

class DomainSubmodule(str, Enum):
    molecules = "molecules"
    properties = "properties"
```
And then:
```suggestion
    domain: DomainSubmodule = Field(
        ..., example=DomaineSubmodule.molecules, description="Submodule of gt4sd.properties"
     )
```
In this way if a string different from the supported ones is provided the class will fail with a validation error.

Reply:
perfect!! I was looking for an argument like `choices` in the `Field` class that would achieve the same but I didnt find anything